src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,48 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (!herbivore.hidden && !(herbivore instanceof Carnivore)) {
+      herbivore.health -= 50;
+
+      if (herbivore.health < 1) {
+        Animal.alive.splice(
+          Animal.alive.indexOf((e) => e.name === herbivore.name),
```

Comment:
`indexOf` takes a value to be found and not a function, perhaps you meant to use `findIndex` instead;
The way this function is defined your `.indexOf` will always return `-1` which will cause your `splice` to always remove the last animal in `Animal.alive` array;

Try this test case:
```js
test(`Herbivores should be removed from 'Animal.alive' array, when their health is 0 (but Zebra is not the last defined animal)`, () => {
    const zebra = new Herbivore('Zebra');
    const tiger = new Carnivore('Tiger');

    tiger.bite(zebra);
    tiger.bite(zebra);

    expect(Animal.alive.includes(zebra))
        .toBe(false);

    expect(Animal.alive.includes(tiger))
        .toBe(true);
});
```